File: java/unidad_02/Problema_2_6/src/Problema_2_6.java
import java.util.Scanner;
import java.util.InputMismatchException;

/*
 * 
 * El programa, dado como dato un número entero, determina si el
 * mismo es par, impar o nulo.
 *
 * @author Angel P.
 * 
 */

public class Problema_2_6 {

    public static void main(String[] args) {

        int numero;

        try (Scanner scanner = new Scanner(System.in)) {

            System.out.print("Número entero: ");
            numero = scanner.nextInt();

        } catch (InputMismatchException e) {

            System.out.println(e);
            return;
        }
        if (Math.pow(-1, numero) > 0) {

            System.out.printf("%s es un número par", numero);
            // -1 elevado a un número par, es positivo
            return;
        }
        System.out.printf("%s es un número impar", numero);
        // -1 elevado a un número impar, es negativo
    }
}

File: python/unidad_02/problema_2_6.py
"""
El programa, dado como dato un número entero, determina si el
mismo es par, impar o nulo.
"""

def main():

    try:

        numero = int(input("Número entero: "))

    except ValueError as e:

        print(e)
        return       
    
    if (- 1) ** numero > 0:

        print(f"{numero} es un número par")
        # -1 elevado a un número par, es positivo
        return

    print(f"{numero} es un número impar")
    # -1 elevado a un número impar, es negativo

if __name__ == '__main__':

    main()